core/state/intra_block_state.go

```diff
@@ -103,6 +111,253 @@ func New(stateReader StateReader) *IntraBlockState {
 		accessList:        newAccessList(),
 		transientStorage:  newTransientStorage(),
 		balanceInc:        map[libcommon.Address]*BalanceIncrease{},
+		readMap:           make(map[blockstm.Key]blockstm.ReadDescriptor),
```

Comment:
please take a look if you can use existing methods: `ibs.SoftFinalise()`, `ibs.MakeWriteSet`, `txTask.ReadLists = rw.stateReader.ReadSet()`, `txTask.WriteLists = rw.bufferedWriter.WriteSet()`. 
you can find them in: 
cmd/state/exec3/state.go

Reply:
Thanks for the pointers. Will look into them.